<?xml version="1.0"?>
<launch>
    <!-- PARAMETERS FOR SETTING UP ARMOR CLIENT-->
    <!-- The path for the ontology to load into ARMOR -->
	<arg name="ontology_path" default="$(find final_assignment)/worlds/topological_map.owl" />
    <!-- The ontology uri associated with the ontology loaded into ARMOR -->
    <arg name="ontology_uri" default="http://bnc/exp-rob-lab/2022-23" />
    <!-- The reference name associated to the ontology loaded into ARMOR -->
    <arg name="ontology_reference_name" default="ontology_reference" />
    <!-- The initial position of the Robot1 -->
    <arg name="robot_room" default="E" />
    
	<!-- LAUNCHING NODE -->
    <node name="map_builder_helper" pkg="final_assignment" type="ontology_map_builder.py" output="screen">
        <param name="ontology_path" type="str" value="$(arg ontology_path)" />
        <param name="ontology_uri" type="str" value="$(arg ontology_uri)" />
        <param name="ontology_reference_name" type="str" value="$(arg ontology_reference_name)" />
    
        <!-- PARAMETERS FOR CREATING THE MAP -->
        <!-- The list of rooms inside the map -->
        <param name="rooms" type="str" value="[
            'R1', 'R2', 'R3', 'R4', 'E', 'C1', 'C2'
        ]"/>
        <!-- The list of doors inside the map -->
        <param name="rooms_doors" type="str" value="[
            ['D1'],
            ['D2'],
            ['D3'],
            ['D4'],
            ['D6', 'D7'],
            ['D1', 'D2', 'D5', 'D6'],
            ['D3', 'D4', 'D5', 'D7']
        ]"/>
        <!-- The positions of the rooms in the space relateive to the world transform -->
        <param name="rooms_positions" type="str" value="[
            [10, 5],
            [15, 5],
            [10, 15],
            [15, 15],
            [5, 10],
            [10, 8],
            [10, 14]
        ]" />
        <param name="robot_room" type="str" value="$(arg robot_room)" />
    </node>
</launch>
